tests/backstop/engine_scripts/puppet/onReady.js

```diff
@@ -1 +1,115 @@
-module.exports = async () => {};
+/**
+ * Custom onReady script for BackstopJS.
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const POLLING_INTERVAL_MS = 100;
+const FONT_WAIT_TIMEOUT_MS = 5000;
+
+/**
+ * Creates a promise that resolves after a specified delay.
+ *
+ * @since n.e.x.t
+ *
+ * @param {number} ms The delay in milliseconds.
+ * @return {Promise<void>} A promise that resolves after the delay.
+ */
+const delay = ( ms ) => new Promise( ( resolve ) => setTimeout( resolve, ms ) );
+
+/**
+ * Executes custom waiting logic before taking screenshots.
+ *
+ * This script applies specific waiting logic based on the scenario label.
+ * Scenarios don't support passing function props therefore waiting logic
+ * is implemented here and triggered via text props.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} page     The Puppeteer page object.
+ * @param {Object} scenario The scenario configuration.
+ * @param {Object} viewport The viewport configuration.
+ */
+module.exports = async ( page, scenario, viewport ) => {
+	// NOTE: We can implement waitForRegistry or other not arbitrary time based delays here to improve test stability for other VRTs.
+
+	// Reset the font size of the specific selector and retrigger resize events.
+	if ( scenario.resetDataBlockGroup ) {
```

Comment:
This is a very brittle workaround to get the VRT tests to pass, which relies on knowledge of the internals of the component. It would be much better to avoid it and treat the component as a black box.

Recognising the time already sunk into this issue, and the value of stabilising our VRT suite, I'm willing to merge it for now, but please create a followup issue where we can investigate and provide a cleaner solution to this problem.

Reply:
This makes sense @techanvil. 
I have created the new issue here: https://github.com/google/site-kit-wp/issues/11057

Reply:
That's great, thanks @ankitrox!